base/loading.jl

```diff
@@ -407,9 +407,24 @@ end
 
 ## generic project & manifest API ##
 
-const project_names = ("JuliaProject.toml", "Project.toml")
-const manifest_names = ("JuliaManifest.toml", "Manifest.toml")
-const preferences_names = ("JuliaLocalPreferences.toml", "LocalPreferences.toml")
+const project_names = ["JuliaProject.toml", "Project.toml"]
+const default_manifest_names = ["JuliaManifest.toml", "Manifest.toml"]
+const preferences_names = ["JuliaLocalPreferences.toml", "LocalPreferences.toml"]
+
+function manifest_names()
+    env_var_name = "JULIA_MANIFEST_NAME"
+    custom_def = strip(get(ENV, env_var_name, ""))
+    if !isempty(custom_def)
+        man_name = expand_version_placeholders(custom_def)
+        if endswith(custom_def, ":")
```

Comment:
Should this split on `:` and replace empty parts with the default like load path?

I also don't think this functionality is that necessary, what is the problem with falling back to `Manifest.toml` if a versioned one doesn't exist?